Controllers films users

Dmitrii-Petrov opened the pull request.

Dmitrii-Petrov committed:
.

Dmitrii-Petrov committed:
(1)

Dmitrii-Petrov committed:
первая итерация ТЗ-9

Dmitrii-Petrov committed:
Delete maven-wrapper.jar

Dmitrii-Petrov committed:
Delete .mvn/wrapper directory

Dmitrii-Petrov committed:
Delete mvnw

Dmitrii-Petrov committed:
Delete mvnw.cmd

Dmitrii-Petrov committed:
первая итерация ТЗ-9

Dmitrii-Petrov committed:
Merge remote-tracking branch 'origin/controllers-films-users' into controllers-films-users

Dmitrii-Petrov committed:
первая итерация ТЗ-9

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/servise/ValidateService.java:
```diff
@@ -0,0 +1,14 @@
+
+public class ValidateService {
+
+    public void validateFilm(Film film) {
```
для удобства можно сделать кастомную аннотацию https://blog.clairvoyantsoft.com/spring-boot-creating-a-custom-annotation-for-validation-edafbf9a97a4

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/servise/ValidateService.java:
```diff
@@ -0,0 +1,14 @@
+
+    public void validateFilm(Film film) {
+        if (film.getReleaseDate().isBefore(LocalDate.of(1895, 12, 28))) {
+            throw new RuntimeException("кино еще не изобрели");
```
лучше кидать кастомную ошибку ValidationException

Dmitrii-Petrov replied:
done

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,40 @@
+    @Pattern(regexp = "[a-zA-Z0-9]*")
+    String login;
+
+    String name;
```
можно максимальный размер задать

Dmitrii-Petrov replied:
done

db-exp commented on src/main/java/ru/yandex/practicum/filmorate/repository/FilmRepository.java:
```diff
@@ -0,0 +1,33 @@
+
+    public void update(Film film) {
+        if (!filmRepository.containsKey(film.getId())) {
+            throw new RuntimeException("фильм не существует");
```
тут тоже можно и лучше сразу делать NotFoundException

Dmitrii-Petrov replied:
поменял на NoSuchElementException

Dmitrii-Petrov committed:
feat: добавить FilmValidation, FilmValidator
fix: исправить прокидываемые ошибки
feat: ограничить максимальный размер поля login в User.java

Dmitrii-Petrov committed:
fix: исправить ограничение максимального размер поля name в User.java
